### PR TITLE
Fixes bumbles resting

### PR DESCRIPTION
**Pull request by Xantholne**
## About The Pull Request

Fixes Bumbles not automatically resting or, really being lively at all since I missed this part

## Why It's Good For The Game

Bumbles will do more than just float

## Changelog
:cl:
fix: Bumbles will now actually rest, sit up, and buzz
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/bumbles.dm [hunk 1]
```diff
@@ -5,10 +5,10 @@
 	icon_living = "bumbles"
 	icon_dead = "bumbles_dead"
 	turns_per_move = 1
-	response_help = "shoos"
+	response_help = "pets"
 	response_disarm = "brushes aside"
 	response_harm = "squashes"
-	speak_emote = list("bzzzs")
+	speak_emote = list("bzzes")
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/friendly/bumbles.dm]: Shouldn't it be "buzzes"?

File: code/modules/mob/living/simple_animal/friendly/bumbles.dm [hunk 2]
```diff
@@ -21,9 +21,9 @@
 	mob_biotypes = MOB_ORGANIC|MOB_BEAST
 	gold_core_spawnable = FRIENDLY_SPAWN
 	verb_say = "bzzs"
-	verb_ask = "bzzs inquisitively"
-	verb_exclaim = "bzzs intensely"
-	verb_yell = "bzzs intensely"
+	verb_ask = "bzzes inquisitively"
+	verb_exclaim = "bzzes intensely"
+	verb_yell = "bzzes intensely"
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/friendly/bumbles.dm:29]:
```suggestion
	emote_see = list("buzzes.", "makes a loud buzz.", "rolls several times.", "buzzes happily.")
	speak_chance = 1

```

File: code/modules/mob/living/simple_animal/friendly/bumbles.dm [hunk 3]
```diff
@@ -44,3 +44,19 @@
...
+			collar_type = "[initial(collar_type)]_rest"
+			set_resting(TRUE)
+		else if (prob(1))
+			if (resting)
+				emote("me", EMOTE_VISIBLE, pick("wakes up with a smiling buzz.", "rolls upside down before waking up.", "stops resting."))
+				icon_state = "[icon_living]"
+				collar_type = "[initial(collar_type)]"
+				set_resting(FALSE)
+			else
+				emote("me", EMOTE_VISIBLE, pick("buzzes.", "makes a loud buzz.", "rolls several times.", "buzzes happily "))
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/friendly/bumbles.dm]:
```suggestion
/mob/living/simple_animal/pet/bumbles/handle_automated_movement()
	. = ..()
	if(!isturf(loc) || !CHECK_MOBILITY(src, MOBILITY_MOVE)  || buckled)
		return
	if(!resting && prob(1))
		emote("me", EMOTE_VISIBLE, pick("curls up on the surface below ", "is looking very sleepy.", "buzzes softly ", "looks around for a flower nap "))
		set_resting(TRUE)
	else if (resting && prob(1))
		emote("me", EMOTE_VISIBLE, pick("wakes up with a smiling buzz.", "rolls upside down before waking up.", "stops resting."))
		set_resting(FALSE)

/mob/living/simple_animal/pet/bumbles/update_mobility()
	. = ..()
	if(stat != DEAD)
		if(!CHECK_MOBILITY(src, MOBILITY_STAND))
			icon_state = "[icon_living]_rest"
		else
			icon_state = "[icon_living]"
		regenerate_icons()

```

**Review by Ghommie — CHANGES_REQUESTED** [review]
No parent call on Life(), bad copypasta, Bumbles not being an actual bee subtype... And I was hoping this silly thing would remain unused while I was busy dealing with other mistakes.

File: code/modules/mob/living/simple_animal/friendly/bumbles.dm (hunk 1 above)
Comment (Xantholne) [reply code/modules/mob/living/simple_animal/friendly/bumbles.dm]: Yeah you're right, meant to put buzzes

**Review by Ghommie — APPROVED** [review]

**Pull request merged**